packages/compiler/src/compiler/base/nodes/tags/message.ts

```diff
@@ -120,7 +120,6 @@ function buildMessage<R extends MessageRole>(
 
     return {
       role,
-      id: String(attributes.id),
       content,
     } as ToolMessage
   }
```

Comment:
Why? A `tool` message requires an `id`, which identifies the `tool_call` that requested it

Reply:
the schema is just wrong this is the valid schemas both for tool result and tool calling requested https://sdk.vercel.ai/docs/reference/ai-sdk-core/stream-text#messages-role not sure where this id thing came from

Reply:
This differs from openai's schema, which is what the compiler is based on
https://platform.openai.com/docs/api-reference/chat/create#chat-create-messages

Do we want to change the compiler's schema to fit vercel's instead?

Reply:
talked offline